processing/src/test/java/org/apache/druid/query/scan/NestedDataScanQueryTest.java

```diff
@@ -463,13 +462,13 @@ public void testExpectedTypes() throws Exception
         NESTED_SPARSE_MIXED_NUMERIC_FIELD
     );
     Assert.assertNotNull(sparseMixedNumericValueSelector);
-    Assert.assertTrue(sparseMixedNumericValueSelector instanceof DimensionDictionarySelector);
+    Assert.assertTrue(sparseMixedNumericValueSelector instanceof ColumnValueSelector);
 
     ColumnValueSelector sparseMixedValueSelector = columnSelectorFactory.makeColumnValueSelector(
         NESTED_SPARSE_MIXED_FIELD
     );
     Assert.assertNotNull(sparseMixedValueSelector);
-    Assert.assertTrue(sparseMixedValueSelector instanceof DimensionDictionarySelector);
+    Assert.assertTrue(sparseMixedValueSelector instanceof ColumnValueSelector);
```

Comment:
I believe that this `instanceof` check would've been true before your changes too.  Perhaps you are wanting to ensure that it's not returning the `DimensionDictionarySelector`-based option, in which case maybe additionally have an

```
Assert.assertFalse(sparseMixedValueSelector instanceof DimensionDictionarySelector);
```